### PR TITLE
Update multiarch docker build

### LINKED ISSUE #6597: Kubernetes Dashboard and Metrics Scrapper on Apple Silicon (Arm64v8) starting AMD64 images
##### Environment
Kubernetes Dashboard 2.4.0
Apple Silicon M1
Docker Desktop

##### Steps to reproduce
Basically just
docker run kubernetesui/dashboard:v2.4.0

##### Observed result
It will allways pull and start an amd64 image.

##### Expected result
It should run an arm64(v8) image

##### Comments
The Problem can be observed directly in docker, as well inside of kubernetes in docker.
Instead of using an arm64 image, docker will always retrieve the wrong AMD64 image and running in emulation.
Even adding the correct DIGEST/SHA256 does not seem to help.

What I also observed is that the architecture type is "linux/arm64", in constrast for example to postgres
where the type is "linux/arm64/v8" and i think Docker expects the additional v8 .. not an expert here though

### PR DESCRIPTION
**Pull request by floreks**
- Use `docker buildx` instead of `docker build` to build correct multiarch images
- Update CD pipeline to properly setup the docker pipeline
- Update `eslint` version and configuration to fix OOM issue during linting the code

Fixes #6597

### CODE REVIEW

**Comment by floreks** [comment]
@goafabric could you check it now by pulling our dev image?
```bash
docker pull kubernetesdashboarddev/dashboard
```

**Comment by maciaszczykm** [comment]
@floreks It doesn't work:

```
➜ docker pull kubernetesdashboarddev/dashboard

Using default tag: latest
Error response from daemon: manifest for kubernetesdashboarddev/dashboard:latest not found: manifest unknown: manifest unknown
```

**Comment by goafabric** [comment]
@floreks @maciaszczykm 
judging from dockerhub .. the tag ist just :head and not :latest

however.. with head still the same problem:

docker run kubernetesdashboarddev/dashboard:head

WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

@floreks 
do you have access to an apple m1 machine to verify ?

**Comment by floreks** [comment]
@goafabric I don't have any apple machine to test on. Please test again but using the `latest` tag.
```bash
docker pull kubernetesdashboarddev/dashboard
```

**Comment by goafabric** [comment]
@floreks 
could not test it inside kubernetes for now
but at least docker run looks good now !

no more complaints / warnings about amd64 image .. so it seems to work

thank you for your really fast changes ...
just keep in mind, that the same is also needed for the metrics scrapper

**Comment by floreks** [comment]
It will still take some time to polish everything and adjust the CI/CD pipeline to auto-deploy correctly built images. I hope to finish everything next week.

**Comment by goafabric** [comment]
> It will still take some time to polish everything and adjust the CI/CD pipeline to auto-deploy correctly built images. I hope to finish everything next week.

just take your time .. as said .. both work just ok on apple silicon,
but i think it will increase stability and startup time 

i'd wish the paketo people would react that fast ...or at all

File: Makefile [hunk 1]
```diff
@@ -232,37 +232,34 @@ e2e-headed: start-cluster
 .PHONY: docker-build-release
 docker-build-release: build-cross
 	for ARCH in $(ARCHITECTURES) ; do \
-  		CGO_ENABLED=0 docker build -t $(RELEASE_IMAGE)-$$ARCH:$(RELEASE_VERSION) -t $(RELEASE_IMAGE)-$$ARCH:latest dist/$$ARCH ; \
-  done
+			docker buildx build \
```
Comment (maciaszczykm) [inline Makefile:235]: Seems like indent is too big.

File: Makefile [hunk 2]
```diff
@@ -232,37 +232,34 @@ e2e-headed: start-cluster
...
+	done ; \
 
 .PHONY: docker-push-release
 docker-push-release: docker-build-release
-	for ARCH in $(ARCHITECTURES) ; do \
-  		docker push $(RELEASE_IMAGE)-$$ARCH:$(RELEASE_VERSION) ; \
-  		docker push $(RELEASE_IMAGE)-$$ARCH:latest ; \
-  done ; \
-  docker manifest create --amend $(RELEASE_IMAGE):$(RELEASE_VERSION) $(RELEASE_IMAGE_NAMES) ; \
+	docker manifest create --amend $(RELEASE_IMAGE):$(RELEASE_VERSION) $(RELEASE_IMAGE_NAMES) ; \
```
Comment (maciaszczykm) [inline Makefile:246]: As above. Please check it.

**Pull request merged**